src/org/zaproxy/zap/extension/api/CoreAPI.java

```diff
@@ -926,13 +930,29 @@ public ApiResponse handleApiView(String name, JSONObject params)
 				((ApiResponseList)result).addItem(new ApiResponseElement("host", site));
 			}
 		} else if (VIEW_SITES.equals(name)) {
-			result = new ApiResponseList(name);
-			SiteNode root = (SiteNode) session.getSiteTree().getRoot();
-			@SuppressWarnings("unchecked")
-			Enumeration<TreeNode> en = root.children();
-			while (en.hasMoreElements()) {
-				((ApiResponseList)result).addItem(new ApiResponseElement("site", ((SiteNode) en.nextElement()).getNodeName()));
-			}
+            result = new ApiResponseList(name);
+            if (!Constant.isLowMemoryOptionSet()) {
```

Comment:
- It's not necessary to differentiate based on low memory option, both should use `StructuralNode` (like in the else but using `SessionStructure` to get the root node, this requires a tweak in `SessionStructure` though e.g. in `find` to return the root node when the URL is `null`).
- It should be used the same indentation as the surrounding code.